src/components/cesium_map/cesiumh3.js

```diff
@@ -111,24 +111,24 @@ export class H3GridManager {
 
     update(cview, rect) {
         const rstr = r2str(rect);
-        console.log(`update: ${rstr}`);
         const existing = cview.dataSources.getByName(rstr);
         if (existing.length > 0) {
             console.log(`Grid ${rstr} already in collection`);
             return;
         }
-        let g = this.global_grid;
         if (rect !== GLOBAL_RECT) {
-            g = new H3Grid();
+            this.global_grid = new H3Grid();
```

Comment:
Changed to directly accessing and modifying the class variable (`this.global_grid`). Somehow the `this.global_grid` value was not getting updated, and after some debugging I learned an interesting thing - assigning values to the variable `g` does not affect the `this.global_grid` property of the class instance, as the new value assigned to the variable `g` overwrites the object `this.global_grid` that `g` was referencing. As result, `this.global_grid` will be still be pointing to the old original memory location even though we change values of `g`.